src/webview/html/messageAsHtml.js

```diff
@@ -72,14 +79,98 @@ $!${messageReactionListAsHtml(reactions, ownUser.user_id, allImageEmojiById)}
 `;
 };
 
-const widgetBody = (message: Message | Outbox) => template`
+/**
+ * Render the body of a message that has submessages.
+ *
+ * Must not be called on a message without any submessages.
+ */
+const widgetBody = (message: Message, ownUserId: UserId) => {
+  invariant(
+    message.submessages !== undefined && message.submessages.length > 0,
+    'should have submessages',
+  );
+
+  const widgetSubmessages: Array<{
+    sender_id: number,
+    content: SubmessageData,
+    ...
+  }> = message.submessages
+    .concat()
+    .sort((m1, m2) => m1.id - m2.id)
+    .filter(submessage => submessage.msg_type === 'widget')
+    .map(submessage => ({
+      sender_id: submessage.sender_id,
+      content: JSON.parse(submessage.content),
+    }));
+
+  const errorMessage = template`
 $!${message.content}
 <div class="special-message"
  ><p>Interactive message</p
  ><p>To use, open on web or desktop</p
 ></div>
 `;
 
+  const pollWidget = widgetSubmessages.shift();
+  if (!pollWidget || !pollWidget.content || pollWidget.content.widget_type !== 'poll') {
+    return errorMessage;
+  }
+
+  // $FlowFixMe[incompatible-type]
+  const pollWidgetContent: WidgetData = pollWidget.content;
```

Comment:
```suggestion
  /* $FlowFixMe[incompatible-type]: The first widget submessage should be
       a `WidgetData`; see jsdoc on `SubmessageData`. */
  const pollWidgetContent: WidgetData = pollWidget.content;
```

When writing a fixme, we always want some kind of comment saying why it's appropriate -- like, what kind of assumption is being expressed and where that assumption comes from.